dot/parachain/backing/per_relay_parent_state.go

```diff
@@ -55,63 +207,99 @@ func (rpState *perRelayParentState) importStatement(
 	}
 
 	// PersistedValidationData should not be nil if the statementVDT is Seconded.
 	if signedStatementWithPVD.PersistedValidationData == nil {
 		return nil, errNilPersistedValidationData
 	}
 
-	paraID := committedCandidateReceipt.Descriptor.ParaID
-
-	if rpState.prospectiveParachainsMode.IsEnabled {
-		chIntroduceCandidate := make(chan error)
-		subSystemToOverseer <- parachaintypes.ProspectiveParachainsMessageIntroduceCandidate{
-			IntroduceCandidateRequest: parachaintypes.IntroduceCandidateRequest{
-				CandidateParaID:           paraID,
-				CommittedCandidateReceipt: committedCandidateReceipt,
-				PersistedValidationData:   *signedStatementWithPVD.PersistedValidationData,
-			},
-			Ch: chIntroduceCandidate,
-		}
-
-		introduceCandidateErr, ok := <-chIntroduceCandidate
-		if !ok {
-			return nil, fmt.Errorf("%w: %s",
-				errRejectedByProspectiveParachains,
-				"Could not reach the Prospective Parachains subsystem.",
-			)
-		}
-		if introduceCandidateErr != nil {
-			return nil, fmt.Errorf("%w: %w", errRejectedByProspectiveParachains, introduceCandidateErr)
-		}
-
-		subSystemToOverseer <- parachaintypes.ProspectiveParachainsMessageCandidateSeconded{
-			ParaID:        paraID,
-			CandidateHash: candidateHash,
-		}
+	// Introduce candidate to the prospective parachain subsystem.
+	if err := rpState.introduceCandidate(
+		subSystemToOverseer, committedCandidateReceipt, signedStatementWithPVD.PersistedValidationData); err != nil {
+		return nil, err
 	}
 
-	// Only save the candidate if it was approved by prospective parachains.
 	perCandidate[candidateHash] = &perCandidateState{
 		persistedValidationData: *signedStatementWithPVD.PersistedValidationData,
-		secondedLocally:         false, // This is set after importing when seconding locally.
-		paraID:                  paraID,
+		secondedLocally:         false,
 		relayParent:             committedCandidateReceipt.Descriptor.RelayParent,
 	}
 
-	return rpState.table.importStatement(&rpState.tableContext, signedStatementWithPVD.SignedFullStatement)
+	return rpState.findCoreIndexAndImportStatement(signedStatementWithPVD)
+}
+
+// introduceCandidate sends a message to the Prospective Parachains subsystem to introduce a candidate.
+func (rpState *perRelayParentState) introduceCandidate(
+	subSystemToOverseer chan<- any,
+	committedCandidateReceipt parachaintypes.CommittedCandidateReceipt,
+	persistedValidationData *parachaintypes.PersistedValidationData,
+) error {
+	paraID := committedCandidateReceipt.Descriptor.ParaID
+
+	chIntroduceCandidate := make(chan error)
+	subSystemToOverseer <- parachaintypes.ProspectiveParachainsMessageIntroduceCandidate{
+		IntroduceCandidateRequest: parachaintypes.IntroduceCandidateRequest{
+			CandidateParaID:           paraID,
+			CommittedCandidateReceipt: committedCandidateReceipt,
+			PersistedValidationData:   *persistedValidationData,
+		},
+		Ch: chIntroduceCandidate,
+	}
+
+	introduceCandidateErr, ok := <-chIntroduceCandidate
```

Comment:
to effectively block and wait on the channel you must not read the `ok` value. 
```go
        // waits until the channel returns some value
	introduceCandidateErr := <-chIntroduceCandidate
	
	// don't wait
	// if there is some value ready in the channel it will read 
	// if there is no value ready then it will only get the ok
        //    - (true if the ch is open, false if the ch is closed)
	// but will not block
        introduceCandidateErr, ok := <-chIntroduceCandidate
```

```suggestion
	introduceCandidateErr := <-chIntroduceCandidate
```

Reply:
I don't think the logic will continue if you read if the chan is open/close, it will still be blocked until next value is ready

Reply:
@EclesioMeloJunior That's not true.
`Ok` will be false only when channel is closed.

if there is no value ready then it will wait to get the value. 
You can check running this code https://go.dev/play/p/5349f7J80ug

Reply:
Oh, okay, thanks for the clarification! 👍